src/feature.ts

```diff
@@ -8,4 +8,5 @@ export interface FeatureInterface {
     stale: boolean;
     strategies: StrategyTransportInterface[];
     variants: VariantDefinition[];
+    variantStickiness?: string;
```

Comment:
I am not too happy about introducing a new field to the feature toggle object, as it clutters the [data return from the api](https://unleash.github.io/docs/api/admin/features#fetching-feature-toggles) a bit. 

In a perfect world the variants would not be an array, but an object, so that we could attach the meta-config directly.

Maybe we can introduce the stickiness information on the variant objects inside the array? 

```json
{
   "name": "variant1",
   "weight": 50,
   "stickiness": "some-custom-field"
},
{
   "name": "variant2",
   "weight": 50,
   "stickiness": "some-custom-field"
}
````

This is not 100% optimal either, as you can not have different variants with different stickiness...

Reply:
> In a perfect world the variants would not be an array, but an object, so that we could attach the meta-config directly.

Yes, I think so as well. I did not want to introduce breaking changes.

I am not really convinced by the array either. Could somebody ever want to define different stickiness?

Reply:
> Could somebody ever want to define different stickiness?

Good question. I'm not sure, but I think the override is more relevant if you need to control who sees what.